cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/batch/AbstractBatchContext.java

```diff
@@ -37,32 +40,45 @@
   private final DatasetContext datasetContext;
   private final long logicalStartTime;
   private final Map<String, String> runtimeArgs;
+  private final Admin datasetAdmin;
 
   protected AbstractBatchContext(PluginContext pluginContext,
                                  DatasetContext datasetContext,
                                  Metrics metrics,
                                  LookupProvider lookup,
                                  String stageName,
                                  long logicalStartTime,
-                                 Map<String, String> runtimeArgs) {
+                                 Map<String, String> runtimeArgs,
+                                 Admin datasetAdmin) {
     super(pluginContext, metrics, lookup, stageName);
     this.datasetContext = datasetContext;
     this.logicalStartTime = logicalStartTime;
     this.runtimeArgs = runtimeArgs;
+    this.datasetAdmin = datasetAdmin;
   }
 
   protected <T extends PluginContext & DatasetContext> AbstractBatchContext(T context,
                                                                             Metrics metrics,
                                                                             LookupProvider lookup,
                                                                             String stageName,
                                                                             long logicalStartTime,
-                                                                            Map<String, String> runtimeArgs) {
+                                                                            Map<String, String> runtimeArgs,
+                                                                            Admin datasetAdmin) {
```

Comment:
better to name Admin as `admin` and not `datasetAdmin` though currently we only have dataset operations in Admin.